File: frontend/types/student.ts
export interface Student {
  id: string;
  tutor_id: string;
  first_name: string;
  last_name: string;
  email?: string | null;
  phone?: string | null;
  subject: string;
  lesson_price: number;
  notes?: string | null;
  status: "active" | "paused" | "finished";
  created_at: string;
  updated_at?: string | null;
}

export interface StudentFilter {
  status?: string;
  search?: string;
  page?: number;
  per_page?: number;
}

export interface StudentListResponse {
  students: Student[];
  total: number;
  page: number;
  per_page: number;
}

export interface StudentStats {
  total: number;
  active: number;
  paused: number;
  finished: number;
}

export interface Lesson {
  id: string;
  student_id: string;
  tutor_id?: string;
  date: string;
  start_time: string;
  end_time: string;
  price: number;
  comment?: string | null;
  status: "scheduled" | "completed" | "cancelled";
  payment_status: "paid" | "unpaid";
  student_name?: string;
  created_at: string;
}

export interface Payment {
  lesson_id: string;
  date: string;
  price: number;
  payment_status: "paid" | "unpaid";
}

export interface CreateStudentData {
  first_name: string;
  last_name: string;
  email: string;
  phone: string;
  subject: string;
  lesson_price: number;
  notes?: string;
}

export interface UpdateStudentData extends Partial<CreateStudentData> {
  status?: Student["status"];
}

File: backend/app/schemas/student.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class StudentCreate(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=100)
    last_name: str = Field(..., min_length=1, max_length=100)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=20)
    subject: str = Field(..., min_length=1, max_length=100)
    lesson_price: float = Field(..., gt=0)
    notes: str | None = None


class StudentUpdate(BaseModel):
    first_name: str | None = Field(None, min_length=1, max_length=100)
    last_name: str | None = Field(None, min_length=1, max_length=100)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=20)
    subject: str | None = Field(None, min_length=1, max_length=100)
    lesson_price: float | None = Field(None, gt=0)
    notes: str | None = None
    status: Literal["active", "paused", "finished"] | None = None


class StudentStatusUpdate(BaseModel):
    status: Literal["active", "paused", "finished"]


class StudentResponse(BaseModel):
    id: UUID
    tutor_id: UUID
    first_name: str
    last_name: str
    email: str | None
    phone: str | None
    subject: str
    lesson_price: float
    notes: str | None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class StudentListResponse(BaseModel):
    students: list[StudentResponse]
    total: int
    page: int
    per_page: int
